src/containerapp/azext_containerapp/custom.py

```diff
@@ -3262,3 +3264,57 @@ def set_registry_job(cmd, name, resource_group_name, server, username=None, pass
     containerapp_job_registry_set_decorator.construct_payload()
     r = containerapp_job_registry_set_decorator.set()
     return r
+
+
+# maintenance config
+def add_maintenance_config(cmd, resource_group_name, env_name, duration, start_hour_utc, weekday):
```

Comment:
Will you support customized name in the future? Current design is not very flexible for extensibility. Consider adding `--name` support for your commands with default value = "default" which will be extensible in the future.

Reply:
Hi @Juliehzl 

https://github.com/Azure/azure-cli-extensions/pull/8190#discussion_r1834706856

only one maintenance config of name "default" is allowed.
If we want to support name in the future, we can add `name` in that time. If we add it now, it will be exposed to customer.

Reply:
@p-bouchon

Reply:
We currently have no plans to allow for non default names for maintenance configuration. If any were to be added, they would be specific non-customizable names to specify upgrades on specific aspects of the service